### ISSUE TITLE
[PP] Tests on ephemeral models

### ISSUE THREAD

**Issue by jtcohen6** [fishtown-analytics/dbt#3435]
### Describe the bug

With partial parsing turned on, executing a test against an ephemeral model trips a mysterious error:

```
$ dbt test
Running with dbt=0.20.0-rc1
Partial parsing enabled: 0 files deleted, 0 files added, 0 files changed.
Partial parsing enabled, no changes found, skipping parsing
Found 1 model, 1 test, 1 snapshot, 0 analyses, 333 macros, 0 operations, 1 seed file, 0 sources, 0 exposures

18:00:02 | Concurrency: 1 threads (target='dev')
18:00:02 |
18:00:02 | 1 of 1 START test unique_ephemeral_model_id.......................... [RUN]
Unhandled error while executing test.my_project.unique_ephemeral_model_id.9493b44128
__enter__
18:00:02 | 1 of 1 ERROR unique_ephemeral_model_id............................... [ERROR in 0.02s]
18:00:02 |
18:00:02 | Finished running 1 test in 5.54s.

Completed with 1 error and 0 warnings:

__enter__

Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
```

Here's the full stacktrace:
```
Traceback (most recent call last):
  File "/Users/jerco/dev/product/dbt/core/dbt/task/base.py", line 348, in safe_run
    result = self.compile_and_execute(manifest, ctx)
  File "/Users/jerco/dev/product/dbt/core/dbt/task/base.py", line 285, in compile_and_execute
    ctx.node = self.compile(manifest)
  File "/Users/jerco/dev/product/dbt/core/dbt/task/compile.py", line 33, in compile
    return compiler.compile_node(self.node, manifest, {})
  File "/Users/jerco/dev/product/dbt/core/dbt/compilation.py", line 466, in compile_node
    node, _ = self._recursively_prepend_ctes(
  File "/Users/jerco/dev/product/dbt/core/dbt/compilation.py", line 317, in _recursively_prepend_ctes
    manifest.sync_update_node(cte_model)
  File "/Users/jerco/dev/product/dbt/core/dbt/contracts/graph/manifest.py", line 592, in sync_update_node
    with self._lock:
AttributeError: __enter__
```

### Steps To Reproduce

**models/ephemeral_model.sql**
```sql
{{ config(materialized='ephemeral') }}

select 1 as id
```

**models/schema.yml**
```yml
version: 2
models:
  - name: ephemeral_model
    columns:
      - name: id
        tests:
          - unique
```

```
$ dbt --no-partial-parse test
$ dbt --partial-parse test
```

### Expected behavior
The test should execute as normal.

### System information
**Which database are you using dbt with?**
- [x] postgres
- [ ] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
$ dbt --version
installed version: 0.20.0-rc1
   latest version: 0.19.1

Your version of dbt is ahead of the latest release!

Plugins:
  - postgres: 0.20.0rc1
  - snowflake: 0.20.0rc1
  - redshift: 0.20.0rc1
  - bigquery: 0.20.0rc1
```

**Comment by gshank** [comment]
My guess is that this is related to the manifest._lock field. It would not serialize, so I created a metadata lambda that was supposed to create a new flags.MP_CONTEXT.Lock object. This is probably not working properly.

**Comment by leahwicz** [comment]
@gshank are we good to close this out based off PR #3445 ?

**Comment by jtcohen6** [comment]
Resolved by #3445
